File: problems/2605_form_smallest_number_from_two_digit_arrays/solution.py
from typing import List


class Solution:
    def minNumber(self, nums1: List[int], nums2: List[int]) -> int:
        s1, s2 = set(nums1), set(nums2)
        common = s1 & s2
        if common:
            return min(common)
        a, b = min(nums1), min(nums2)
        return min(a * 10 + b, b * 10 + a)

File: problems/2605_form_smallest_number_from_two_digit_arrays/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int minNumber(vector<int>& nums1, vector<int>& nums2) {
        set<int> s1(nums1.begin(),nums1.end()), s2(nums2.begin(),nums2.end());
        for (int n : s1) if (s2.count(n)) return n;
        int a = *s1.begin(), b = *s2.begin();
        return min(a*10+b, b*10+a);
    }
};
